File: lms-backend/src/main/java/com/lms/entity/Quiz.java
package com.lms.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "quizzes")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Quiz {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

	@Column(nullable = false)
	private String title;

	@Column(columnDefinition = "TEXT")
	private String description;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "course_id", nullable = false)
	private Course course;

	@Column(nullable = false)
	private Integer totalQuestions;

	@Column(nullable = false)
	private Integer totalMarks;

	@Column(nullable = false)
	private Integer timeLimit; // in minutes

	@Column(nullable = false)
	private LocalDateTime startTime;

	@Column(nullable = false)
	private LocalDateTime endTime;

	@Column(name = "created_at", nullable = false, updatable = false)
	private LocalDateTime createdAt;

	@Column(name = "updated_at")
	private LocalDateTime updatedAt;

	@PrePersist
	protected void onCreate() {
		createdAt = LocalDateTime.now();
		updatedAt = LocalDateTime.now();
	}

	@PreUpdate
	protected void onUpdate() {
		updatedAt = LocalDateTime.now();
	}
}

File: lms-backend/src/models/Quiz.js
const { DataTypes } = require("sequelize");

module.exports = (sequelize) => {
  return sequelize.define("Quiz", {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    title: { type: DataTypes.STRING, allowNull: false },
    description: { type: DataTypes.TEXT },
    courseId: { type: DataTypes.INTEGER },
    totalQuestions: { type: DataTypes.INTEGER, defaultValue: 10 },
    totalMarks: { type: DataTypes.INTEGER, defaultValue: 20 },
    timeLimit: { type: DataTypes.INTEGER, defaultValue: 15 },
    questions: { type: DataTypes.JSON, defaultValue: [] },
  });
};
